### ISSUE TITLE
Package binaries from relative path

### ISSUE THREAD

**Issue by sturmf** [conan-io/conan#1680]
I have the following project structure after the compilation process

build/x86/Debug/*.dll
build/x86/Release/*.dll
build/x64/Debug/*.dll
build/x86/Release/*.dll
source/xy/inc/*.h
source/xy/*.cpp
source/conanfile.py

and I now want to package the dlls and header files into a neat conan package.
Unfortunately I struggle writing a proper conan file. It seems that the method of packaging existing binaries with only a package_info step does not work since for that the header and libs need to be in a proper directory structure next to each other already. Therefore I tried writing a build step which copies the files to a lib and include directory. But here I now face the problem that i would need to copy from e.g. ..\build\x86\Debug\* but while the recipie is running I am already in the .conan\data\... directory and my relative path no longer works.

It also seems that I can't hand in the current directory as --cwd

Any ideas how to package the binaries? And I do not want to build the whole package, everything is already on disk at that moment.

**Comment by memsharded** [comment]
I think that you could be very interested in trying the RC for 0.26, which includes functionality to use the ``package()`` method while executing ``conan package_files``, so it will copy just the necessary files.

Please try it installing with: ``pip install conan==0.26.0rc1``

Then you have to implement the ``package()`` method, using the ``self.settings.build_type`` and ``self.settings.arch`` to package one or the other files, then use the command:

````bash
$ conan package_files Pkg/version@user/channel -pr=your_profile --build_folder=<your-folder>
```

**Comment by sturmf** [comment]
Hey I think this works the way I want it. My package step now looks like this:

    def package(self):
        platform = "x64" if self.settings.arch == "x86_64" else "x32"
        self.copy("*.dll", dst="lib", src="%s/export/%s" % (platform, self.settings.build_type) , keep_path=False)

**Comment by memsharded** [comment]
I think you need to fix it, I don't know what the "export" folder accounts for. You also might need to package the headers and the linkage libraries. Without testing, I would try something like:

```python
def package(self):
      platform = "x64" if self.settings.arch == "x86_64" else "x32"
      folder = "%s/%s" %  (self.settings.build_type, platform)
      #usually dlls go to the bin folder, not the lib one. You might skip keep_path, if not nested
      self.copy("*.dll", dst="bin", keep_path=False)
      #using "include" you don't need to specify it in package_info(), it is the default
      self.copy("*.h", dst="include", src="xy/inc")
      # if in windows, you probably want to package linkage .libs too
      ...
```

A good idea is to setup a repo I can try, like this one:
https://github.com/memsharded/simple_package_files_example

**Comment by sturmf** [comment]
My full solution now looks like the following, the exports folder is just an extra folder we have in our output directory hierarchy.

    def package(self):
        platform = "x64" if self.settings.arch == "x86_64" else "x86"
        folder = "%s/export/%s" %  (platform, self.settings.build_type)
        if self.options.shared:
            self.copy("*.dll", dst="bin", src=folder)
            self.copy("*.lib", dst="lib", src=folder, excludes="*_static.lib")
        else:
            self.copy("*_static.lib", dst="lib", src=folder)
        self.copy("*.h", dst="include", src="../include")
        
    def package_info(self):
        self.cpp_info.libs = self.collect_libs()

does this sound sensible? If yes I think we can close this issue.

**Comment by memsharded** [comment]
Yes, looks good. This feature has just been released in conan 0.26, so please update and try. Also, it is very easy to check if your packages are looking exactly as you want, the conan cache is just a folder, and is very easy to navigate. So you can go to ``<userhome>/.conan/data/Pkg/version/user/channel/packages/shaXXXX`` and you should see there your package with your final layout.

Also, the ``test_package`` functionality is great, very, very recommended if you are creating packages. So you can have a ``test_package`` folder besides your conanfile.py, and fire the test with:

```bash
$ conan test_package Pkg/version@user/channel --test_only
```
That is the definitive check that the package is OK. Closing the issue, please ask if you have any further question. Thanks!
